src/util/codec/number.rs

```diff
@@ -219,6 +222,149 @@ pub trait NumberDecoder: Read {
 
 impl<T: Read> NumberDecoder for T {}
 
+type Bytes<'a> = &'a [u8];
+
+#[inline]
+fn read_num_bytes<T, F>(size: usize, data: &mut &[u8], f: F) -> Result<T>
```

Comment:
Honestly I don't think it's a good idea to provide post-transform function in this function. As the name indicates, it should read bytes, but in fact it will return other type of things. In addition, this post process is not hard / complex to implement in callers. Are there any reasons to merge them together?

Reply:
The inspiration comes from https://docs.rs/byteorder/1.2.2/src/byteorder/lib.rs.html#1767.
And if we do not merge them together, the same code block would be duplicated again and again.

Reply:
more in detail?

Reply:
I think is ok to use `f` here just like https://docs.rs/byteorder/1.2.2/src/byteorder/lib.rs.html#1767. If not  we should consider changing the name `read_num_bytes`